From 3d98f4a156223a9e6734cf0f1606be0d2873fa49 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Wed, 11 Oct 2017 15:53:38 -0500
Subject: [PATCH 2639/5242] MLK-16571-3: arm64: dts: add i.MX8QM lpspi device
 node

commit  a06bef6d251bb06c316a5b64b4c7a47c695b6da9 from
https://source.codeaurora.org/external/imx/linux-imx.git

add the lpspi device node and change the peripheral to nor chip.
i.MX8QM also need both ipg and per clock for this module.

BuildInfo: SCFW 9e9f6ec6, IMX-MKIMAGE 0, ATF 0

Reviewed-by: Pan Gao <pandy.gao@nxp.com>
Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts  |   21 ++++++++++++++------
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi      |    7 ++++---
 2 files changed, 19 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
index 394d03a..06972c1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
@@ -409,11 +409,16 @@
 				SC_P_SPI0_SCK_DMA_SPI0_SCK		0x0600004c
 				SC_P_SPI0_SDO_DMA_SPI0_SDO		0x0600004c
 				SC_P_SPI0_SDI_DMA_SPI0_SDI		0x0600004c
-				SC_P_SPI0_CS0_DMA_SPI0_CS0		0x0600004c
 				SC_P_SPI0_CS1_DMA_SPI0_CS1		0x0600004c
 			>;
 		};
 
+		pinctrl_lpspi0_cs: lpspi0cs {
+			fsl,pins = <
+				SC_P_SPI0_CS0_LSIO_GPIO3_IO05		0x21
+			>;
+		};
+
 		pinctrl_flexspi0: flexspi0grp {
 			fsl,pins = <
 				SC_P_QSPI0A_DATA0_LSIO_QSPI0A_DATA0	0x0600004c
@@ -719,15 +724,19 @@
 &lpspi0 {
 	#address-cells = <1>;
 	#size-cells = <0>;
+	fsl,spi-num-chipselects = <1>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lpspi0>;
+	pinctrl-0 = <&pinctrl_lpspi0 &pinctrl_lpspi0_cs>;
+	cs-gpios = <&gpio3 5 GPIO_ACTIVE_LOW>;
 	status = "okay";
 
-	spidev0: spi@0 {
+	flash: at45db041e@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "atmel,at45", "atmel,dataflash";
+		spi-max-frequency = <500000>;
 		reg = <0>;
-		compatible = "rohm,dh2228fv";
-		spi-max-frequency = <4000000>;
-	};
+        };
 };
 
 &lpuart0 { /* console */
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index a10d1ad..7d9081d5 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -1865,10 +1865,11 @@
 		reg = <0x0 0x5a000000 0x0 0x10000>;
 		interrupts = <GIC_SPI 216 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-parent = <&gic>;
-		clocks = <&clk IMX8QM_SPI0_CLK>;
-		clock-names = "ipg";
+		clocks = <&clk IMX8QM_SPI0_CLK>,
+			 <&clk IMX8QM_SPI0_IPG_CLK>;
+		clock-names = "per", "ipg";
 		assigned-clocks = <&clk IMX8QM_SPI0_CLK>;
-		assigned-clock-rates = <32000000>;
+		assigned-clock-rates = <20000000>;
 		power-domains = <&pd_dma_lpspi0>;
 		status = "disabled";
 	};
-- 
1.7.9.5

